src/psyclone/psyir/transformations/parallel_loop_trans.py

```diff
@@ -83,24 +85,30 @@ def validate(self, node, options=None):
 
         :param node: the node we are checking.
         :type node: :py:class:`psyclone.psyir.nodes.Node`
-        :param options: a dictionary with options for transformations.\
-                        This transform supports "collapse", which is the\
-                        number of nested loops to collapse.
+        :param options: a dictionary with options for transformations.
         :type options: Optional[Dict[str, Any]]
-        :param int options["collapse"]: number of nested loops to collapse
-                                        or None.
+        :param bool|int options["collapse"]: if it's a bool and is False
+            (default), it won't collapse. If it's a bool and is True, it will
+            collapse as much as possible. If it's an integer, it will attempt
+            to collapse until the specified number of loops (if they exist and
+            are safe to collapse them). The options 'ignore_dependencies_for'
+            and 'force' also affect the collapse applicabilty analysis.
         :param bool options["force"]: whether to force parallelisation of the
-                target loop (i.e. ignore any dependence analysis).
+            target loop (i.e. ignore any dependence analysis).
+        :param list[str] options["ignore_dependencies_for"]: whether to ignore
+            some symbol names from the dependence analysis checks.
         :param bool options["sequential"]: whether this is a sequential loop.
+        :param bool options["verbose"]: whether report the reasons the validate
+            and collapse steps have failed.
 
-        :raises TransformationError: if the \
-                :py:class:`psyclone.psyir.nodes.Loop` loop iterates over \
-                colours.
-        :raises TransformationError: if 'collapse' is supplied with an \
-                invalid number of loops.
-        :raises TransformationError: if there is a data dependency that \
-                prevents the parallelisation of the loop unless \
-                `options["force"]` is True.
+        :raises TransformationError: if the given loop iterates over a
+                colours (LFRic domain) iteration space.
+        :raises TransformationError: if 'collapse' is not an int or a bool.
```

Comment:
As you mention in #2668, I think this should be a TypeError, not a TransformationError.

Reply:
Done